ch13/ex13_34_36_37.cpp

```diff
@@ -48,7 +44,15 @@ void Message::remove_from_Folders()
 {
     for (auto f : folders)
         f->remMsg(this);
-    folders.clear();
+    // The book added one line here: folders.clear(); but I think it is redundant and more importantly, it will cause a bug:
+    // - In Message::operator=, in the case of self-assignment, it first calls remove_from_Folders() and its folders.clear() 
+    //   clears the data member of lhs(rhs), and there is no way we can assign it back to lhs.
+    //   Refer to: http://stackoverflow.com/questions/29308115/protection-again-self-assignment
+    // - Why is it redundant? As its analogous function Message::add_to_Folders(), Message::remove_from_Folders() should ONLY
+    //   take care of the bookkeeping in Folders but not touch the Message's own data members - makes it much clearer and easier
+    //   to use. As you can see in the 2 places where we call Message::remove_from_Folders(): in Message::operator=, folders.clear()
+    //   introduces a bug as illustrated above; in the destructor ~Message(), the member "folders" will be destroyed anyways, why do
+    //   we need to clear it first?
```

Comment:
Sorry for my poor understanding, could you please show the code for reappear the bug your mentioned?